compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1101,20 +1101,88 @@ private STNode parseFuncDefOrFuncTypeDesc(STNode metadata, STNode visibilityQual
         return funcDefOrType;
     }
 
-    private STNode parseFunctionKeywordRhs(STNode metadata, STNode visibilityQualifier, List<STNode> qualifiers,
-                                           STNode functionKeyword, boolean isObjectMember, boolean isObjectTypeDesc) {
-        switch (peek().kind) {
-            case IDENTIFIER_TOKEN:
+    /**
+     * Parse function definition for the function type descriptor.
+     * <p>
+     * <code>
+     * function-defn := `function` identifier function-signature function-body
+     * <br/>
+     * function-type-descriptor := `function`
+     * </code>
```

Comment:
This doc is inconsistent with the method. Let's remove the code lines and change the top line to something like `Parse function definition or function type descriptor identifier rhs`. Les's change the method name accordingly